Source/santactl/Commands/SNTCommandRule.m

```diff
@@ -45,6 +46,49 @@ + (NSString *)shortHelpText {
   return @"Manually add/remove/check rules.";
 }
 
+- (BOOL)isWellformedSHA256:(NSString *)hash {
+  if (hash.length != 64) {
+    return NO;
+  }
+
+  NSCharacterSet *hexChars =
+    [NSCharacterSet characterSetWithCharactersInString:@"0123456789ABCDEF"];
+
+  return [@"" isEqualToString:[hash stringByTrimmingCharactersInSet:hexChars]];
```

Comment:
I think it'll be quicker in the average case to filter out the inverted set and _then_ check the length == 64.

Reply:
If I'm understanding correctly, I think this could leave invalid data. E.g. the string `AA...Z...AA` might return a length of 64 after being trimmed, but the charset is still invalid

Reply:
@russellhancox So you mean something like the following?

```objective-c
- (BOOL)isWellformedSHA256:(NSString *)hash {
NSCharacterSet *nonHexChars =
    [[NSCharacterSet characterSetWithCharactersInString:@"0123456789ABCDEF"] invertedSet];

  return [hash stringByTrimmingCharactersInSet:nonHexChars].length == 64;
}
```

Reply:
Yeah, perfect

Reply:
`stringByTrimmingCharactersInSet` only trims from the front and back of a string. The proposed method doesn't properly verify strings like the following, that start and end with valid chars, but have invalid chars in the middle:
```
[self isWellformedSHA256:@"A!KLM(DSNSINNCIOSDNMLKSDM<SDM><><S>D<F>SD<F<EWIOJFEWJCOIWECNM./A"];
```
This returns `true`, but obviously not desired.

Reply:
Ah, good point. I don't know if NSString has a better way but this would work:

```objc
	NSCharacterSet *nonHexChars =
	    [[NSCharacterSet characterSetWithCharactersInString:@"0123456789abcdef"] invertedSet];
	hash = [[hash componentsSeparatedByCharactersInSet:nonHexChars] componentsJoinedByString:@""];
	return hash.length == 64;
```

Reply:
Seems like this would probably work. But also I think this would require at least 2 copies and additional allocations. I don't think this code is on a hot path so (??) so it maybe doesn't matter, but the original impl is likely more performant. 

If this is hot path and we do care about speed, it might be worth abstracting the code in `WatchItems.mm` that checks for valid hex strings of a given length in one pass without extra allocations/copies: https://github.com/google/santa/blob/main/Source/santad/DataLayer/WatchItems.mm#LL89

Reply:
This isn't on a hot path but we should put that somewhere in common and re-use it, so we don't have multiple implementations of the same thing. I'm sure there are other places it could (or should) be used.